src/shapleypy/classes/checkers.py

```diff
@@ -0,0 +1,61 @@
+# ruff: noqa: N806
+from __future__ import annotations
+
+from shapleypy.coalition import (
+    EMPTY_COALITION,
+    Coalition,
+    all_one_player_missing_subcoalitions,
+)
+from shapleypy.game import Game
+
+
+def check_monotonicity(game: Game) -> bool:
+    """
+    Check if the game is monotone.
+    """
+    for S in game.all_coalitions:
+        # Check the values of all subcoalitions of size |S| - 1
+        value_of_S = game.get_value(S)
+        for S_minus_one in all_one_player_missing_subcoalitions(S):
+            if value_of_S < game.get_value(S_minus_one):
+                return False
+    return True
+
+
+def check_weakly_superadditivity(game: Game) -> bool:
+    grand_coalition = Coalition.grand_coalition(game.number_of_players)
+    for i in range(game.number_of_players):
+        value_of_i = game.get_value([i])
+        for S in (grand_coalition - i).all_subcoalitions():
+            if game.get_value(S) + value_of_i > game.get_value(S + i):
+                return False
+    return True
+
+
+def check_superadditivity(game: Game) -> bool:
+    for T in game.all_coalitions:
+        value_of_T = game.get_value(T)
+        for S in filter(
+            lambda s: T * s == EMPTY_COALITION, game.all_coalitions
+        ):
+            if value_of_T + game.get_value(S) > game.get_value(T + S):
+                return False
+    return True
+
+
+def check_convexity(game: Game) -> bool:
+    grand_coalition = Coalition.grand_coalition(game.number_of_players)
+    for i in range(game.number_of_players):
+        for T in (grand_coalition - i).all_subcoalitions():
+            i_contribution_to_T = game.get_value(T + i) - game.get_value(T)
+            for S in T.all_subcoalitions():
+                if (
+                    game.get_value(S + i) - game.get_value(S)
+                    > i_contribution_to_T
+                ):
+                    return False
+    return True
+
+
+def check_supermodularity(game: Game) -> bool:
+    return check_convexity(game)
```

Comment:
We should implement checker for positive games (sometimes called totally monotonic). Those are exactly those games, defined as  non-negative linear combinations of unanimity games (see https://kam.mff.cuni.cz/~cerny/data/CG/CG4-shapleyho-hodnota_handout.pdf for unanimity games).

Positive games are interesting thanks to their strong properties and also we might want to do a generator for them because it is easy to do so (from mathematical point of view, at least).

Reply:
Added in https://github.com/muzikr/Shapleypy/pull/21/commits/d358b6e6b317220992c12e4ac15fbdd1814fe36b